### ISSUE TITLE
Get Browser Console Logs  support

### ISSUE THREAD

**Issue by tomaspekarovic** [MarketSquare/robotframework-browser#2176]
Because lot of information is shown in console log, it would be great if there is any chance how to capture this logs.

**Solution**
Maybe it should be feasible to have something like  'recordHar'  when opening context to save logs to file, or even to  expose GLOBAL variable with console logs?

**Alternatives**
I tried to use custom js keyword, promises etc but nothing was successful. Maybe there is some way but I did not found it yet.

I also tried to fire promise or some kind of 'background' task (using google, because I'm not Js developer) but nothing was able to capture logs.

**Comment by aaltat** [comment]
@allcontributors please add @tomaspekarovic for ideas

**Comment by tomaspekarovic** [comment]
ok I found out, that 
```
async function openNewPage(url, page, logger) {
  page.on('console', msg => logger("console text: " + msg.text()));
  await page.goto(url);
  await page.waitForTimeout(15000);
}
```
Can be used to handle logs, but the problem is, that  `page.on` event does not start to listening on every console event. It just works only in context of the keyword.

So keyword is listening for 15 s. Maybe the solution is to somehow enable to send `on` events to page when the page is created?

```
    New Page    https://google.sk    on_event=(console, msg => logger('console text: ' + msg.text()))
    function    msg => logger('console text: ' + msg.text())
    New Page    https://google.sk    on_event=(console, function)
```
Maybe even to set more  'on_event' hooks to the page. This would definitely helps not only for console logs, but also for automatic handling of events that can occur on the page.

I know that I can create JS keywords but the on event will be only in the scope of keyword. If it will be in the general scope, I can handle popups, console logs, and everything that playwright offers.

Also maybe there is any keyword/possibility to 'return' that page object and apply it in general ?  maybe just to apply that event?

**Comment by aaltat** [comment]
`page.on` event should work in background, example we already set `close` [event](https://github.com/MarketSquare/robotframework-browser/blob/main/node/playwright-wrapper/playwright-state.ts#L211) and it works nicely. Also `console` are already [collected](https://github.com/MarketSquare/robotframework-browser/blob/main/node/playwright-wrapper/playwright-state.ts#L196-L198). So in theory, one should write a keyword which collects the console messages from `consoleMessages` list.

**Comment by tomaspekarovic** [comment]
Thanks Tatu, at the end I found out that there is way how to get console logs. 
Keyword   `Close Page` will return console logs and error logs right after the page is closed.  I will try to get inspired and if nobody will be faster than me I'll try to create PR (hopefully)

**Comment by mkorpela** [comment]
As this is already possible with `Close Page` I'll take the high prio away.

**Comment by jin-harmoney** [comment]
I would like to expand on this request:

`Close Page` indeed returns console logs, but we're relying on automatic page closing. So my next idea was to use an [initialization file](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#initialization-files) and add the `Close Page` keyword in the `Test Teardown`. Unfortunately, this is a `default` teardown and is not executed when there is a teardown specified in the test case. Seems there is no way to run a teardown on every test in robot framework at this moment. I found a ticket that requested this feature, but it was rejected in favor of using Listeners.

I also tried a `jsextension` with `page.on('console', ...)`, but didn't get it to work.

The only solution I see at this moment is to add `Close Page` in every teardown, but that's tedious ofcourse. I also tried [Listeners](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#listener-interface), but I couldn't figure out how to connect to the running browser instance..
